lib/vrl/compiler/src/compiler.rs

```diff
@@ -440,6 +440,12 @@ impl<'a> Compiler<'a> {
             None => (None, None),
         };
 
+        // Keep track of the known scope *before* we compile the closure.
+        //
+        // This allows us to revert to any known state that the closure
+        // arguments.might overwrite.
```

Comment:
```suggestion
        // Keep track of the known scope *before* we compile the closure.
        //
        // This allows us to revert to any known state that the closure
        // arguments might overwrite.
```

I think that's a typo and we don't have a `might` argument? :trollface: